### PR TITLE
feat: Ensure enum fields are received as string values over REST transport

### PR DESCRIPTION
**Pull request by vchudnov-g**
Requests now error if enum fields are not represented as strings.

This PR includes compliance tests.

### CODE REVIEW

**Comment by noahdietz** [comment]
I will take a look tomorrow (4/14)!

File: util/genrest/resttools/checkrestbody.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+	}
+
+	fieldName = strings.Join(nameParts, ".")
+	if !found {
+		return fieldName, true
+	}
+
+	if !isString {
+		return fieldName, false
+	}
```
Comment (noahdietz) [inline util/genrest/resttools/checkrestbody.go:98]: Can you add some more supporting commentary in here? I got a little thrown by the loop control logic and how it determined what was being evaluated once the loop exited and got to these checks.

**Review by noahdietz — APPROVED** [review]
Just nits, LGTM

File: util/genrest/resttools/checkrestbody.go [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+	if _, err := strconv.Atoi(value); err == nil {
```
[the remark this answers is not in the document]
Comment (vchudnov-g) [reply util/genrest/resttools/checkrestbody.go:100]: Nah, we're good. The enums in natural and synthetic proto3s will fit into `int32`

https://developers.google.com/protocol-buffers/docs/proto3#enum
https://developers.google.com/protocol-buffers/docs/proto#enum
`Enumerator constants must be in the range of a 32-bit integer.`

Note also that we're not checking for numeric encoding in other bases. This is an extra-helpful check, but the main check is for the JSON type above.

File: util/genrest/resttools/checkrestbody.go (hunk 1 above)
Comment (vchudnov-g) [reply util/genrest/resttools/checkrestbody.go:98]: Added some comments. WDYT?

File: util/genrest/resttools/checkrestbody.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (noahdietz) [reply util/genrest/resttools/checkrestbody.go:105]: Sounds good! Thanks for the links TIL

**Review by noahdietz — APPROVED** [review]
LGTM

**Pull request merged**